onnx/test/cpp/function_verify_test.cc

```diff
@@ -49,51 +45,53 @@ void VerifyTypeConstraint(
     const OpSchema* schema = OpSchemaRegistry::Schema(
         op_type, function_op.since_version(), function_op.domain());
 
-    std::unordered_map<std::string, int> input_tensor_name_idx_map;
-    std::unordered_map<std::string, int> output_tensor_name_idx_map;
-    // Enforce it on input
-    for (unsigned int i = 0; i < schema->inputs().size(); ++i) {
-      auto& input = schema->inputs().at(i);
-      input_tensor_name_idx_map[input.GetName()] = i;
-    }
-    for (auto& tensor_name_tc : tc_map) {
-      auto iter = input_tensor_name_idx_map.find(tensor_name_tc.first);
-      if (iter == input_tensor_name_idx_map.end())
-        continue;
-      const auto& types = schema->inputs().at(iter->second).GetTypes();
-      std::unordered_set<std::string> allowed_types;
-      for (auto& s : types) {
-        allowed_types.insert(*s);
-      }
-      for (auto& type : tensor_name_tc.second) {
-        if (allowed_types.find(type) == allowed_types.end()) {
-          fail_check(
-              "Input type " + type + " defined in " + schema->Name() +
-              "'s function body is not allowed in node " + op_type);
+    // Check inputs of node:
+    parameter_index num_formal_inputs =
+        static_cast<parameter_index>(schema->inputs().size());
+    parameter_index num_actual_inputs = node.input_size();
+    // The above may not be same due to optional inputs
+    auto num_checked_inputs = std::min(num_formal_inputs, num_actual_inputs);
```

Comment:
Is there a case where actual inputs > formal input?

Reply:
No. I can add a check and throw an error in this case, if that's what you mean.

Reply:
What I mean why not just use actual inputs?

Reply:
Actually: sorry, I was wrong. This is possible with variadic parameters. I will modify the logic to do the right thing.